pb/pstore.proto

```diff
@@ -23,5 +23,15 @@ message AddrBookRecord {
 
 		// The original TTL of this address.
 		int64 ttl = 3;
+
+		// Whether this address was certified by the addressed peer.
+		bool certified = 4;
 	}
+
+	// The Seq counter value from the most recent RoutingState record for this peer.
+	uint64 routingStateSeq = 3;
+
+	// A serialized SignedEnvelope containing the most recent RoutingState record.
+	// TODO: import the SignedEnvelope type instead
+	bytes signedRoutingRecord = 4;
```

Comment:
So, instead of duplicating everything, we can just decode with _two_ different protobufs.

```protobuf
// Used by the peerstore to find addresses for a given peer.
message AddrEntry {
  message SignedRoutingRecord {
    PublicKey publicKey = 1;
    RoutingState contents = 3;
  }
  SignedRoutingRecord record = 1;
  repeated AddressEntry additionalAddresses = 2;
}
// Use to pick out the signed record (for the DHT).
message OpaqueAddrEntry {
  bytes record = 1;
}
```

Reply:
That's an interesting technique.